test/runnable/bit.d

```diff
@@ -0,0 +1,21 @@
+
+struct S
+{
+    uint x : 2, y : 3, :0;
+    int :0;
```

Comment:
They all use the same code to resolved bit fields, and the same code that ImportC uses. Not even copied code, it's the same code in the same functions.

Reply:
The whole point of this PR was to use the same code as was tested in ImportC.

Reply:
That may be the case now but it still needs to be tested since it might not be the case in the future.

Reply:
The whole point of mentioning classes and non-pod structs is to ensure that it works with druntime GC.

The whole point of mentioning `__traits` is because C doesn't have reflection.

We aren't testing this currently with ImportC.

Reply:
ImportC *does* have a CTFE engine. It came for free from using the D semantic engine. I'd have had to deliberately *break* it for it not to work. The same goes for the other points.

BitfieldDeclaration is derived from VarDeclaration, and the only overriding it does is layout. Otherwise, everything is the same.

Reply:
That is based on a flawed assumption - the type of a bitfield is essentially a lie because the behaviour differs from the base type due to the limited size. This causes misbehaviour e.g. [for overflows](https://issues.dlang.org/show_bug.cgi?id=22710)

Reply:
I checked the internal type that gdc generates for bit-fields, and it correctly sets the backend properties of ie: `int a:2;` to `.min = -2, .max = 1`.  Haven't checked the D front-end though.

Reply:
Ok, I added these tests, except for the template one where I don't know what you mean.

Reply:
> Ok, I added these tests, except for the template one where I don't know what you mean.

```
struct S(T)
{
    T t : 3;
}
S!int good;
S!float bad;
```